env0/resource_environment.go

```diff
@@ -627,6 +627,31 @@ func updateTemplate(d *schema.ResourceData, apiClient client.ApiClientInterface)
 
 func deploy(d *schema.ResourceData, apiClient client.ApiClientInterface) diag.Diagnostics {
 	deployPayload := getDeployPayload(d, apiClient, true)
+
+	subEnvironments, err := getSubEnvironments(d)
+	if err != nil {
+		return diag.Errorf("failed to extract subenvrionments from resourcedata: %v", err)
+	}
+
+	if len(subEnvironments) > 0 {
+		deployPayload.SubEnvironments = make(map[string]client.SubEnvironment)
+
+		for i, subEnvironment := range subEnvironments {
+			configuration := d.Get(fmt.Sprintf("sub_environment_configuration.%d.configuration", i)).([]interface{})
+			configurationChanges := getConfigurationVariablesFromSchema(configuration)
+			configurationChanges = getUpdateConfigurationVariables(configurationChanges, subEnvironment.Id, apiClient)
+
+			for i := range configurationChanges {
+				configurationChanges[i].Scope = client.ScopeEnvironment
+			}
+
+			deployPayload.SubEnvironments[subEnvironment.Alias] = client.SubEnvironment{
+				Revision:             subEnvironment.Revision,
+				ConfigurationChanges: configurationChanges,
+			}
+		}
+	}
+
```

Comment:
looks good.
Just one question, where do make sure this code only executes for workflow environments?
As regular environments don't have the `subEnvironments` payload.

Reply:
If subenvironments are used, I now pass the "workflow" type:
bfd059632207e15afaff0ab9625fe2545b14255c

I can always check the template as well (that it's the workflow type). I'm unsure if we would want to go that far. (I would hope the API would fail?)

Reply:
cool, but I don't see you check it in `deploy` before calling `getSubEnvironments`